### PR TITLE
[REVIEW] gpuCI Scripts Update

### PR DESCRIPTION
**Pull request by dillon-cullinan**
**Overview**
First step before "Project Flash" to improve gpuCI scripts and allow for use of different node types for CPU builds. Post "Project Flash" there will be another round of improvements and updates with the goal to make these scripts easier to maintain.

**Changes**
* Consolidates some of the CPU gpuCI scripts, reducing number of scripts to maintain
* Standardizes naming of the upload script to upload.sh
* Changes logging to use gpuci_logger where practical
* Updates comments and documentation where necessary
* Enables gpuCI to override environment variables based on node types

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: ci/cpu/upload.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+export LIBCLX_FILE=`conda build conda/recipes/libclx --output`
+export CLX_FILE=`conda build conda/recipes/clx --python=$PYTHON --output`
+
+################################################################################
+# UPLOAD - Conda packages
+################################################################################
+
+gpuci_logger "Starting conda uploads"
+
+if [ "$BUILD_LIBCLX" == "1" && "$UPLOAD_LIBCLX" == "1" ]; then
```
Comment (mike-wendt) [inline ci/cpu/upload.sh:41]: This needs to be surrounded by `[[ ]]` due to the use of the &&

Same for the one below

File: ci/gpu/build.sh [hunk 2]
```diff
@@ -59,36 +55,84 @@ conda install -y -c pytorch -c gwerbin \
     "matplotlib" \
     "faker"
```
Comment (mike-wendt) [inline ci/gpu/build.sh:61]: Can you add the comment and commands to remove the env pkgs and install a custom pkg version?

**Review by mike-wendt — CHANGES_REQUESTED** [review]
Just some minor fixes. Need to figure out why conda builds are erroring

**Review by raydouglass — APPROVED** [review]

File: ci/cpu/upload.sh (hunk 1 above)
Comment (dillon-cullinan) [reply ci/cpu/upload.sh]: Hm, I copy pasted this file from other PRs, so the syntax should be consistent, and it worked for the others. Problem might lie elsewhere, I'll check.

**Review by mike-wendt — APPROVED** [review]

File: ci/docs/build.sh [hunk 3]
```diff
@@ -16,40 +16,45 @@ export DOCS_WORKSPACE=$WORKSPACE/docs
...
 
-logger "Activate conda env..."
-source activate rapids
-conda install --freeze-installed -c rapidsai-nightly -c rapidsai -c nvidia -c pytorch -c conda-forge \
+gpuci_logger "Activate conda env"
+. /opt/conda/etc/profile.d/conda.sh
+conda activate rapids
+
+gpuci_conda_retry install --freeze-installed -c rapidsai-nightly -c rapidsai -c nvidia -c pytorch -c conda-forge \
     pytorch torchvision requests yaml python-confluent-kafka python-whois markdown beautifulsoup4 jq
```
Comment (ajschmidt8) [inline ci/docs/build.sh:37]:
```suggestion
    pytorch torchvision requests yaml python-confluent-kafka python-whois
```

A few of these can be removed.

**Review by raydouglass — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Pull request merged**